src/components/CardGridSearch.astro

```diff
@@ -0,0 +1,255 @@
+						`;
+					heading.insertAdjacentElement('afterend', wrn);
+				});
+				// Select all "nothing found" created before
+				const warnings = document.querySelectorAll('[id^="warn-"]');
+
+				// Prevent page reload
+				form.addEventListener('submit', (e) => {
+					e.preventDefault();
+				});
+
+				// Prevent reload on 'Enter', and enable clean input on 'Escape'
+				form.addEventListener('keydown', (e) => {
+					clear.classList.remove('pagefind-ui__suppressed');
+					if (e.key === 'Escape') {
+						reset();
+					} else if (e.key === 'Enter') e.preventDefault();
+				});
+				searchBar.addEventListener('keydown', (e) => {
+					if (e.key === 'Enter') e.preventDefault();
+				});
+
+				// Search
+				searchBar.addEventListener('keyup', () => {
+					searchFilter();
+				});
+				// Clear search button 'X'
+				clear.addEventListener('click', () => {
+					clear.classList.add('pagefind-ui__suppressed');
+					reset();
+				});
+
+				function reset() {
+					searchBar.value = '';
+					searchFilter();
+				}
+
+				/**
+				 * Function to search and filter based on input
+				 */
+				function searchFilter() {
+					const query = searchBar.value.toUpperCase();
+
+					inputs.forEach((elements, i) => {
+						const cards = Array.from(elements.children);
+						let total = cards.length;
+
+						cards.forEach((el) => {
+							const htmlElement = el as HTMLElement;
+							const span = el.querySelector('span');
+							const txtValue = span?.textContent || span?.innerText;
+							if (!txtValue) return;
+
+							// Filter list
+							if (txtValue.toUpperCase().indexOf(query) > -1) {
+								total++;
+								htmlElement.style.display = '';
+							} else {
+								total--;
+								htmlElement.style.display = 'none';
+							}
+
+							// Show warning if none found
+							if (total === 0) {
+								warnings[i].classList.remove('hidden');
+							} else {
+								warnings[i].classList.add('hidden');
+							}
+						});
+					});
+				}
+			});
+		}
+	}
+	customElements.define('card-search', CardSearch);
+</script>
+
+<style>
+	:root {
+		/* Starlight */
+		--pagefind-ui-primary: var(--sl-color-accent-light);
+		--pagefind-ui-text: var(--sl-color-gray-2);
+		--pagefind-ui-font: var(--__sl-font);
+		--pagefind-ui-background: var(--sl-color-black);
+		--pagefind-ui-border: var(--sl-color-gray-5);
+		--pagefind-ui-border-width: 1px;
+		--sl-search-cancel-space: 5rem;
+		/* default */
+		--pagefind-ui-scale: 0.8;
+		--pagefind-ui-tag: #eeeeee;
+		--pagefind-ui-border-radius: 8px;
+		--pagefind-ui-image-border-radius: 8px;
+		--pagefind-ui-image-box-ratio: 3 / 2;
+		--pagefind-ui-font: system, -apple-system, 'BlinkMacSystemFont', '.SFNSText-Regular',
+			'San Francisco', 'Roboto', 'Segoe UI', 'Helvetica Neue', 'Lucida Grande', 'Ubuntu', 'arial',
+			sans-serif;
+	}
+	@media (min-width: 50rem) {
+		:root {
+			--sl-search-cancel-space: 0px;
+		}
+	}
+	.pagefind-ui {
+		width: 100%;
```

Comment:
Could we make this span all the way across to match the width of any `LinkCards` below it on smaller viewports? See this screenshot for an example:


![image](https://github.com/tauri-apps/tauri-docs/assets/15347255/b86a5e5d-6c20-4a83-b3dd-bc4f7a0e862f)